app/controllers/admin/impersonates_controller.rb

```diff
@@ -29,19 +39,23 @@ def load_user
     def check_self_impersonation
       if params[:impersonated_user_id] == current_user.id.to_s
         flash[:warning] = "You cannot impersonate yourself"
-        redirect_to URI(request.referer).path
+        redirect_to URI(referer).path
       end
     end
 
     def check_admin_user_impersonation
       if @user.admin?
         flash[:warning] = "You cannot impersonate another admin user"
-        redirect_to URI(request.referer).path
+        redirect_to URI(referer).path
       end
     end
 
     def pundit_user
       true_user
     end
+
+    def referer
+      request.headers["HTTP_REFERER"]
```

Comment:
I'm happy to give this a go but [it looks like it's doing the same as `request.referer`](https://github.com/rack/rack/blob/f5666bc8cb13b8d731ea0222fbd3ada670f2cd55/lib/rack/request.rb#L204).

```ruby
      def referer;         get_header('HTTP_REFERER')                     end
```

Reply:
yeah, agreed, I'll close this PR